src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java

```diff
@@ -1031,17 +1030,16 @@ public void sync(String name) {
   public void unsync(ServoControl sc) {
     if (sc == null) {
       log.error("{}.unsync(null)", getName());
+      return;
     }
     unsync(sc.getName());
   }
 
   @Override
   public void waitTargetPos() {
-    //
-    // while (this.pos != this.targetPos) {
-    // Some sleep perhaps?
-    // TODO:
-    // }
+     while (this.getCurrentInputPos() != this.targetPos) {
```

Comment:
How does this work for servos that don't have access to the raw encoder?

Reply:
All servos currently have access to a time encoder ... heh, its the cheapest and least accurate sensor ever .. but it doesn't require additional hardware and provides the estimate of position at any given time.  "Real" hardware can substitute a real encoder instead of a time estimation - that's kinda what @kwatters DiyServo did